File: code-samples/websockets/migration-before.php
<?php
use RingCentral\SDK\Subscription\Events\NotificationEvent;
use RingCentral\SDK\Subscription\Subscription;

$rcsdk = new RingCentral\SDK\SDK( $_ENV['RC_APP_CLIENT_ID'],
                                  $_ENV['RC_APP_CLIENT_SECRET'],
                                  $_ENV['RC_SERVER_URL'] );
$platform = $rcsdk->platform();
$platform->login( [ "jwt" => $_ENV['RC_USER_JWT'] ] );

$subscription = $rcsdk->createSubscription('Pubnub');
$subscription->addEvents(array('/restapi/v1.0/account/~/extension/~/message-store/instant?type=SMS'));
$subscription->addListener(Subscription::EVENT_NOTIFICATION, function (NotificationEvent $e) {
    print 'Notification ' . print_r($e->payload(), true) . PHP_EOL;
});
$subscription->setKeepPolling(true);
$subscription->register();
?>

File: code-samples/websockets/migration-before.cs
using RingCentral;
using dotenv.net;

DotEnv.Load();

namespace SubscriptionExample {
  class Program {
    static RestClient restClient;
    static PubNubExtension pubnub;
    static async Task Main(string[] args) {
      restClient = new RestClient(
      Environment.GetEnvironmentVariable("RC_APP_CLIENT_ID"),
      Environment.GetEnvironmentVariable("RC_APP_CLIENT_SECRET"),
      Environment.GetEnvironmentVariable("RC_SERVER_URL"));
      await restClient.Authorize(Environment.GetEnvironmentVariable("RC_USER_JWT"));
      pubnub = new PubNubExtension();
      await restClient.InstallExtension(pubnub);
    }
    static private async Task subscribe_to_events() {
      await pubnub.Subscribe(new string[] {
        "/restapi/v1.0/account/~/extension/~/message-store"
        }, message => { Console.WriteLine(message) });
      }
    }
  }

  await rc.Revoke();
